packages/angular_devkit/build_angular/src/builders/jest/index.ts

```diff
@@ -0,0 +1,147 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { BuilderContext, BuilderOutput, createBuilder } from '@angular-devkit/architect';
+import { execFile as execFileCb } from 'child_process';
+import * as path from 'path';
+import { promisify } from 'util';
+import { buildEsbuildBrowserInternal } from '../browser-esbuild';
+import { BrowserEsbuildOptions } from '../browser-esbuild/options';
+import { OutputHashing } from '../browser-esbuild/schema';
+import { normalizeOptions } from './options';
+import { Schema as JestBuilderSchema } from './schema';
+import { findTestFiles } from './test-files';
+
+const execFile = promisify(execFileCb);
+
+/** Main execution function for the Jest builder. */
+export default createBuilder(
+  async (schema: JestBuilderSchema, context: BuilderContext): Promise<BuilderOutput> => {
+    context.logger.warn(
+      'NOTE: The Jest builder is currently EXPERIMENTAL and not ready for production use.',
+    );
+
+    const options = normalizeOptions(schema);
+    const testOut = 'dist/test-out'; // TODO(dgp1130): Hide in temp directory.
```

Comment:
Agreed, my longer-term thought is to generate these files in a temporary directory outside the project and then delete it afterwards to make this less observable. For now it's useful for debugging purposes and since we don't have a `--debug` flag yet this is easiest way to introspect what's going on at runtime. Once we have better debug support, I agree we should hide the directory altogether.